--- mksh/Build.sh.omv~	2015-08-20 03:51:53.790554176 +0200
+++ mksh/Build.sh	2015-08-20 03:52:47.373708471 +0200
@@ -770,6 +770,9 @@ Linux)
 	*) add_cppflags -D_GNU_SOURCE ;;
 	esac
 	add_cppflags -DSETUID_CAN_FAIL_WITH_EAGAIN
+	# We don't have a controlling TTY in some chroot jails...
+	# ... and ABF containers, causing make check to fail
+	add_cppflags -DMKSH_DISABLE_TTY_WARNING
 	: "${HAVE_REVOKE=0}"
 	;;
 LynxOS)
